torch/onnx/utils.py

```diff
@@ -390,10 +390,37 @@ def _create_jit_graph(model, args, _retain_param_name):
             freezed_m = torch._C._freeze_module(model._c, preserveParameters=True)
             module, params = torch._C._jit_onnx_list_model_parameters(freezed_m)
             method_graph = module._get_method('forward').graph
-
-            in_vars, in_desc = torch.jit._flatten(tuple(args) + tuple(params))
-            graph = _propagate_and_assign_input_shapes(
-                method_graph, tuple(in_vars), False, False)
+            print(method_graph)
+
+            has_packed_params = any([True if "PackedParams" in str(input_.type()) else False
+                                    for input_ in method_graph.inputs()])  # type: ignore
+            has_tuple = any([True if input_.type().kind() == "TupleType" else False
+                            for input_ in method_graph.inputs()])  # type: ignore
+
+            if has_packed_params and has_tuple:
+                raise RuntimeError("It is not supported when both packed params and tuples are graph inputs.")
+
+            if has_packed_params:
+                args_params = tuple(args) + tuple(params)
+                # Only torch.Tensor object should be flattened.
+                # If we flatten a tuple (packed param), we will have more in_vars than graph inputs.
+                args_params_tensor_flag = [1 if isinstance(t, torch.Tensor) else 0 for t in args_params]
+                args_params_new = tuple([t for t in args_params if isinstance(t, torch.Tensor)])
+                in_vars, in_desc = torch.jit._flatten(args_params_new)
+                # Use the empty tensor to replace tuple (packed param) as in_vars_new.
+                in_vars_new = [torch.tensor([])] * len(args_params_tensor_flag)
+                idx_old = 0
+                for idx in range(len(args_params_tensor_flag)):
+                    if args_params_tensor_flag[idx] == 1:
+                        in_vars_new[idx] = in_vars[idx_old]
+                        idx_old += 1
```

Comment:
Is there a possibility that the idx_old will be out of range of in_vars?

Reply:
No, should be `len(args_params_new)`